CLAUDE.md

```diff
@@ -0,0 +1,339 @@
+│   ├── Umbraco.PublishedCache.HybridCache/ # Content caching
+│   ├── Umbraco.Examine.Lucene/            # Search indexing
+│   ├── Umbraco.Cms.Persistence.EFCore/    # EF Core data access
+│   ├── Umbraco.Cms.Persistence.EFCore.Sqlite/
+│   ├── Umbraco.Cms.Persistence.EFCore.SqlServer/
+│   ├── Umbraco.Cms.Persistence.Sqlite/    # Legacy SQLite
+│   ├── Umbraco.Cms.Persistence.SqlServer/ # Legacy SQL Server
+│   ├── Umbraco.Cms.Imaging.ImageSharp/    # Image processing v1
+│   ├── Umbraco.Cms.Imaging.ImageSharp2/   # Image processing v2
+│   ├── Umbraco.Cms.StaticAssets/          # Embedded assets
+│   ├── Umbraco.Cms.DevelopmentMode.Backoffice/
+│   ├── Umbraco.Cms.Targets/               # NuGet targets
+│   └── Umbraco.Cms/                       # Meta-package
+│
+├── tests/                                  # 6 test projects
+│   ├── Umbraco.Tests.Common/
+│   ├── Umbraco.Tests.UnitTests/
+│   ├── Umbraco.Tests.Integration/
+│   ├── Umbraco.Tests.Benchmarks/
+│   ├── Umbraco.Tests.AcceptanceTest/
+│   └── Umbraco.Tests.AcceptanceTest.UmbracoProject/
+│
+├── templates/                              # Project templates
+│   └── Umbraco.Templates/
+│
+├── tools/                                  # Build tools
+│   └── Umbraco.JsonSchema/
+│
+├── umbraco.sln                            # Main solution file
+├── Directory.Build.props                  # Shared build configuration
+├── Directory.Packages.props               # Centralized package versions
+├── .editorconfig                          # Code style
+└── .globalconfig                          # Roslyn analyzers
+```
+
+### Architecture Layers
+
+**Dependency Flow** (unidirectional, always flows inward):
+
+```
+Web.UI → Web.Common → Infrastructure → Core
+                ↓
+          Api.Management → Api.Common → Infrastructure → Core
+                ↓
+          Api.Delivery → Api.Common → Infrastructure → Core
+```
+
+**Key Principle**: Core has NO dependencies (pure contracts). Infrastructure implements Core. Web/APIs depend on Infrastructure.
+
+### Project Dependencies
+
+**Core Layer**:
+- `Umbraco.Core` → No dependencies (only Microsoft.Extensions.*)
+
+**Infrastructure Layer**:
+- `Umbraco.Infrastructure` → `Umbraco.Core`
+- `Umbraco.PublishedCache.*` → `Umbraco.Infrastructure`
+- `Umbraco.Examine.Lucene` → `Umbraco.Infrastructure`
+- `Umbraco.Cms.Persistence.*` → `Umbraco.Infrastructure`
+
+**Web Layer**:
+- `Umbraco.Web.Common` → `Umbraco.Infrastructure` + caching + search
+- `Umbraco.Web.UI` → `Umbraco.Web.Common` + all features
+
+**API Layer**:
+- `Umbraco.Cms.Api.Common` → `Umbraco.Web.Common`
+- `Umbraco.Cms.Api.Management` → `Umbraco.Cms.Api.Common`
+- `Umbraco.Cms.Api.Delivery` → `Umbraco.Cms.Api.Common`
+
+---
+
+## 3. Teamwork & Collaboration
+
+### Branching Strategy
+
+- **Main branch**: `main` (protected)
+- **Branch naming**:
+  - Features: `feature/description` or `contrib/username/description`
+  - Bug fixes: `bugfix/description` or `fix/issue-number`
```

Comment:
This isn't correct/complete - see https://dev.azure.com/umbraco/D-Team%20Tracker/_wiki/wikis/D-Team-Tracker.wiki/2435/Branch-Naming